ArduCopter/RC_Channel.cpp

```diff
@@ -583,6 +574,13 @@ bool RC_Channel_Copter::do_aux_function(const aux_func_t ch_option, const AuxSwi
             }
             break;
 
+        case AUX_FUNC::ARMDISARM_WITH_AIRMODE:
+            RC_Channel::do_aux_function_armdisarm(ch_flag);
+            if (copter.arming.is_armed()) {
+                copter.ap.armed_with_airmode_switch = true;
```

Comment:
Shouldn't this have } else { .. = false ?

Reply:
its always set false on arming, see  `ArduCopter/AP_Arming.cpp` above